src/main.cpp

```diff
@@ -3929,6 +3935,8 @@ bool ProcessMessages(CNode* pfrom)
 
         if (!fRet)
             LogPrintf("ProcessMessage(%s, %u bytes) FAILED\n", strCommand.c_str(), nMessageSize);
+        
+        break;
     }
```

Comment:
We don't really need this while loop any longer since we break out after the first iteration of it....